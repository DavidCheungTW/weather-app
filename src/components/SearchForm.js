import React from "react";
import PropTypes from "prop-types";

function SearchForm({ searchText, setSearchText, onSubmit }) {
  // function SearchForm({ onCitySearch }) {
  // const city = document.querySelector("#selectCity").value;
  // const searchButton = document.querySelector(".searchButton");
  // searchButton.addEventListener("click", onCitySearch(city));

  const handleInputChange = (event) => setSearchText(event.target.value);
  // const handleInputChange = (event) => console.log(event.target.value);

  return (
    <div className="search-form">
      <input type="text" onChange={handleInputChange} value={searchText} />
      <button type="submit" onClick={onSubmit}>
        Search
      </button>
    </div>
  );
}

export default SearchForm;

SearchForm.propTypes = {
  searchText: PropTypes.string.isRequired,
  setSearchText: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
};
