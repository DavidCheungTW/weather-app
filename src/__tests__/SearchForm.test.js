import React from "react";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import SearchForm from "../components/SearchForm";

describe("SearchForm", () => {
  const validProps = {
    searchText: "Manchester",
    setSearchText: jest.fn(),
    onSubmit: jest.fn(),
  };

  test("Renders as expected", () => {
    const rendered = renderer.create(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.onSubmit}
      />
    );

    expect(rendered).toMatchSnapshot();
  });

  test("Assert textBox is present", () => {
    render(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.onSubmit}
      />
    );

    expect(screen.getByRole("textbox")).toBeTruthy();
    expect(screen.getByRole("textbox")).toBeInstanceOf(HTMLInputElement);
  });

  test("Assert button is present", () => {
    render(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.onSubmit}
      />
    );
    const buttons = screen.getAllByRole("button");

    expect(buttons).toHaveLength(1);
    expect(buttons[0]).toHaveTextContent("Search");
  });
});
