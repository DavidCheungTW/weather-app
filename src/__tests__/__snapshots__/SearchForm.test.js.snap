// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm Renders as expected 1`] = `
<div
  className="search-form"
>
  <input
    onChange={[Function]}
    type="text"
    value="Manchester"
  />
  <button
    onClick={[MockFunction]}
    type="submit"
  >
    Search
  </button>
</div>
`;
